sentinel-dashboard/src/main/java/com/alibaba/csp/sentinel/dashboard/controller/AuthorityRuleController.java

```diff
@@ -119,10 +111,8 @@
     }
 
     @PostMapping("/rule")
-    public Result<AuthorityRuleEntity> apiAddAuthorityRule(HttpServletRequest request,
-                                                           @RequestBody AuthorityRuleEntity entity) {
-        AuthUser authUser = authService.getAuthUser(request);
-        authUser.authTarget(entity.getApp(), PrivilegeType.WRITE_RULE);
+    @AuthAction(PrivilegeType.WRITE_RULE)
+    public Result<AuthorityRuleEntity> apiAddAuthorityRule(@RequestBody AuthorityRuleEntity entity) {
```

Comment:
> If to the JSON format request body does the `app` parameter still can be fetched from parameters in this action?

yes, it can be fetched

Reply:
> For a simple improvement it's okay. My proposal is to use some `shared parameters` / `path variables` in each request for additional check, in future maybe.
> 
> Now maybe we should support `EL Expression` for simple solution. Please refer to [1], and code example below:
> 
> ```java
> private static final CacheExpressionParser parser = new CacheExpressionParser();
> .....
> 
> String target = (String) parser.parse(targetKey, annotedMeothodKey, method, args);
> ```
> 
> [1] https://github.com/yoloho/enhanced-data/blob/master/enhanced-data-cache/src/main/java/com/yoloho/enhanced/cache/support/CacheExpressionParser.java

I don’t think it’s necessary to be so complicated.

Reply:
It it can be fetched from `{app:"demo"}` then it will not be a problem. Because what you mentioned beyond my understanding so I did an experiment but got negative result as below:

Request(JSON format)
<img width="450" alt="Screen Shot 2019-10-10 at 14 05 49" src="https://user-images.githubusercontent.com/17573073/66543192-48613600-eb67-11e9-862c-20dbd0e0659b.png">

Breakpoint
![Screen Shot 2019-10-10 at 14 05 10](https://user-images.githubusercontent.com/17573073/66543212-5dd66000-eb67-11e9-961c-ee7f46596ea9.png)


So it was a kind of incorrect result here.
One of the solution is simply introducing el expression support to attribute `target`.
And I indeed know due to my latency this PR has been hanged for `quite a long time` while its `original intent` didn't include such scenario.

So should we continue to make it more considerate or go on without this condition considered temporarily? @sczyh30 
Or what do you think about it? @lkxiaolou

Reply:
Oh, sorry, I got it wrong. It cannot be fetched from json.

Reply:
@jasonjoo2010 
```
 /**
     * Represents the current user.
     */
    interface AuthUser {

        /**
         * Query whether current user has the specific privilege to the target, the target
         * may be an app name or an ip address, or other destination.
         * <p>
         * This method will use return value to represent  whether user has the specific
         * privileges to the target, but to throw a RuntimeException to represent no auth
         * is also a good way.
         * </p>
         *
         * @param target        the target to check
         * @param privilegeType the privilege type to check
         * @return if current user has the specific privileges to the target, return true,
         * otherwise return false.
         */
        boolean authTarget(String target, PrivilegeType privilegeType);
```
 The target may be an app name or an ip address, or other destination.

Reply:
It's just an assumption in comment but app name in actual codes only. 

In my opinion we'd better make it clear attributes like `clientIp`, `appName` in annotation and just use appName nowadays. Surely client address can be easily fetched through `request` so it's not necessary to `AuthAction`. 

For this PR, changes related to new introduced annotation is enough. We just need to make new codes `clear` and `correct`.

So can we reach any consensus on JSON-format request and naming?

Reply:
ping for any progress?

or shall we enhanced it later and approve this? @sczyh30 
Only bean type parameter remained now.

Reply:
I'll handle this issue after 11.11 :)

Reply:
We could improve the design to support JSON parsing later. Contributions are welcomed!